bitcoin/src/consensus/serde.rs

```diff
@@ -443,8 +444,16 @@ impl<E: fmt::Debug, I: Iterator<Item = Result<u8, E>>> IterReader<E, I> {
 }
 
 impl<E: fmt::Debug, I: Iterator<Item = Result<u8, E>>> Read for IterReader<E, I> {
-    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+    fn read(&mut self, mut buf: &mut [u8]) -> io::Result<usize> {
         let mut count = 0;
+        if let Some(first) = self.buf.take() {
+            if buf.is_empty() {
+                return Ok(0);
+            }
```

Comment:
You forgot to move this out.

Reply:
Holly shit, might have been quicker to do this PR yourself. Sorry bro, sloppy work.